CONTRIBUTING.md

```diff
@@ -13,6 +9,7 @@
 - Install the latest version of git (https://git-scm.com/downloads)
 - Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell), version 6 or higher, if you plan to make public API changes or are working with generated code snippets.
 - Install [NodeJS](https://nodejs.org/) (16.x.x) if you plan to use [C# code generation](https://github.com/Azure/autorest.csharp).
+- [Fork the repository](https://docs.github.com/get-started/quickstart/fork-a-repo); work will be done in your fork and a pull request should be opened against the `main` branch of the Azure SDK for .NET repository when ready for review.
```

Comment:
I'd go so far as to say a topic branch should be opened as well. You don't want your `origin/main` to differ from `upstream/main` - especially when we squash merge - or you're going to have a bad time. You could link to a sibling doc on GitHub: https://docs.github.com/en/get-started/quickstart/github-flow#create-a-branch (either that bookmark, or just the whole page).

Reply:
Added mention of a topic branch and linked in your suggested doc as well as one for opening PRs from a fork.